Include/internal/pycore_frame.h

```diff
@@ -156,22 +159,29 @@ _PyFrame_GetLocalsArray(_PyInterpreterFrame *frame)
     return frame->localsplus;
 }
 
-/* Fetches the stack pointer, and sets stacktop to -1.
-   Having stacktop <= 0 ensures that invalid
-   values are not visible to the cycle GC.
-   We choose -1 rather than 0 to assist debugging. */
+/* Fetches the stack pointer, and sets stackpointer to NULL.
+   Having stackpointer == NULL ensures that invalid
+   values are not visible to the cycle GC. */
 static inline PyObject**
 _PyFrame_GetStackPointer(_PyInterpreterFrame *frame)
 {
-    PyObject **sp = frame->localsplus + frame->stacktop;
-    frame->stacktop = -1;
+#ifndef Py_DEBUG
```

Comment:
Shouldn't this (and the corresponding line in the next function) be `#ifdef Py_DEBUG`? I don't think the assertion added to `_Py_Dealloc` will ever see an updated value of `sp_cached` otherwise.

Reply:
Thanks.